.github/workflows/ci.yml

```
@@ -99,12 +97,14 @@ jobs:
         env:
           # -Zrandomize-layout makes sure we dont rely on the layout of anything that might change
           RUSTFLAGS: -Zrandomize-layout
+          # https://github.com/rust-lang/miri#miri--z-flags-and-environment-variables
           # -Zmiri-disable-isolation is needed because our executor uses `fastrand` which accesses system time.
           # -Zmiri-ignore-leaks is needed because running bevy_ecs tests finds a memory leak but its impossible
           # to track down because allocids are nondeterministic.
-          # -Zmiri-tag-raw-pointers is not strictly "necessary" but enables a lot of extra UB checks relating
-          # to raw pointer aliasing rules that we should be trying to uphold.
-          MIRIFLAGS: -Zmiri-disable-isolation -Zmiri-ignore-leaks -Zmiri-tag-raw-pointers
+          # -Zmiri-permissive-provenance enables extra UB checks relating to raw pointer aliasing rules that we
+          # should be trying to uphold.
```

That's correct currently but will be wrong in a few days.
Once https://github.com/rust-lang/miri/pull/2275 lands, all this does is disable warnings against int2ptr casts.
So in preparation for that:
```suggestion
          # -Zmiri-permissive-provenance disables warnings against int2ptr casts (since those are used by once_cell)
```